.gitlab-ci.yml

```diff
@@ -0,0 +1,51 @@
+stages:
+  - deploy_to_sonatype
+  - create_key
+
+variables:
+  REGISTRY: 486234852809.dkr.ecr.us-east-1.amazonaws.com
+
+# From the tagged repo, push the release artifact
+deploy_to_sonatype:
+  stage: deploy_to_sonatype
+
+  rules:
+    # Build tags automatically
+    - if: '$CI_COMMIT_BRANCH == "master" && $CI_COMMIT_TAG =~ /^\d+.\d+.\d+/'
```

Comment:
Not sure we want to restrict to the `master` branch? (I think we want any "stable" tag to trigger a release, whatever branch it's on?)

Reply:
I guess maybe I was a bit concerned about security here - if someone makes a malicious branch without a review and then pushes a tag for that branch's commit, they can force a bad release. Since master is usually protected from direct pushes, this seemed like a slight mitigation for that and the manual trigger can be used for non-master builds. I can definitely remove this branch check if the tag triggers seem more reasonable for releasing.

Reply:
ok, but overall at the moment anyone with write access to the repo can push directly to master as well, so not sure this mitigates much (?)

Maybe as a larger mitigation you could make this step _always_ manual, so that access to the internal gitlab is required to trigger releases?

(I don't have a strong opinion to be honest)